theseus/geometry/utils/utils.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import threading
+from typing import Any
+
+import torch
+
+
+class LieGroupContext(object):
+    contexts = threading.local()
+
+    @classmethod
+    def get_context(cls):
+        if not hasattr(cls.contexts, "use_lie_tangent"):
+            cls.contexts.use_lie_tangent = False
+        return cls.contexts.use_lie_tangent
+
+    @classmethod
+    def set_context(cls, use_lie_tangent: bool):
+        cls.contexts.use_lie_tangent = use_lie_tangent
+
+
+class set_lie_tangent_enabled(object):
```

Comment:
Could we rename these as `set_tangent_enabled`, `enable_tangent`, `no_tangent`? (dropping the "lie"). Is that unambiguous enough? Since these context will be used a lot, I'd prefer shorter names.

cc: @mhmukadam

Reply:
My concern is that the Euclidean gradient is mathematically on the Euclidean tangent space, while the Lie tangent gradient is on the Lie group tangent space. Both of them are the tangent spaces.

Reply:
Ah, OK, that makes sense. Then it's probably safest to keep the "lie" prefix.